src/node_crypto.cc

```diff
@@ -1644,9 +1743,12 @@ void SSLWrap<Base>::VerifyError(const FunctionCallbackInfo<Value>& args) {
     x509_verify_error = SSL_get_verify_result(w->ssl_);
   }
 
-  if (x509_verify_error == X509_V_OK)
+  if ((x509_verify_error == X509_V_OK) && CheckWhiteList(w->ssl_))
     return args.GetReturnValue().SetNull();
 
+  // CheckWhiteList might change x509_verify_error. Check it again.
+  x509_verify_error = SSL_get_verify_result(w->ssl_);
```

Comment:
Is `SSL_set_verify_result(ssl, X509_V_ERR_CERT_REVOKED)` in `IsWhitelistedServerCertificate()` also not needed? If not, I fear the inconsistency between `ssl->verify_result` and  `x509_verify_error `.